as/cNFT/assembly/mint.ts

```diff
@@ -1,30 +1,32 @@
 import { context, logging, u128 } from 'near-sdk-as'
 import { assert_deposit_attached } from './utils/asserts'
-import { TokenMetadata } from '../../NFT/assembly/metadata'
-import { Token } from './models/token'
+import { Token } from './models/persistent_tokens'
 import { persistent_tokens } from './models/persistent_tokens'
+import { persistent_tokens_metadata, TokenMetadata } from './models/persistent_tokens_metadata'
 
 @nearBindgen
 export function mint(tokenMetadata: TokenMetadata): Token {
-    /** TODO Assert deposit attached based on custom amount from init */
+    /**@todo Assert deposit attached based on custom amount from init */
     //     assert_deposit_attached(u128.from(0))
 
-    /** TODO Assert uniqueId is actually unique */
-
     let token = new Token()
 
-    token.media = tokenMetadata.media
-    token.extra = tokenMetadata.extra
-    /** TODO not always */
+    /**@todo Not always sender is creator i guess, would rather pass it to function */
```

Comment:
If we pass the creator to mint, won't this make it easy to fake tokens? 
Someone can mint a token using the name of a popular creator

Reply:
Updated the comment. This is just a skeleton, left the comment as a reminder. You can see [here](https://github.com/near-examples/nft-tutorial/blob/4093906440bf8688d1762e8b988688fcd5be5145/nft-contract/src/mint.rs#L6) on approach to minting. Feel free to research others and discuss them on GitHub.